Userland/Libraries/LibWeb/MimeSniff/Resource.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2023, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include "MimeType.h"
+
+namespace Web::MimeSniff {
+
+// https://mimesniff.spec.whatwg.org/#resource
+class Resource {
+public:
+    enum class SniffingContext {
+        None,
+        Browsing,
+        Image,
+        AudioOrVideo
+    };
+
+    static ErrorOr<Resource> create(ReadonlyBytes data, SniffingContext sniffing_context = SniffingContext::None, StringView scheme = ""sv, Optional<MimeType> supplied_type = {}, bool no_sniff = false);
```

Comment:
Went with this approach. I did have an [alternative ](https://github.com/SerenityOS/serenity/pull/21350#discussion_r1382629205) but I feel this makes more sense instead of having callers use 2 statements to get what they need